### PR TITLE
Merge with transactions

### PR DESCRIPTION
**Pull request by AndreaScorza**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        
+        {
+            "type": "pwa-chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (jasny) [inline .vscode/launch.json:16]: Should be in .gitingore.

File: src/LTO.ts [hunk 2]
```diff
@@ -56,11 +41,27 @@ export class LTO {
     }
   }
 
+  public generateNewSeed(length): string {
```
Comment (jasny) [inline src/LTO.ts:44]:
```suggestion
  public generateNewSeed(words: number = 15): string {
```

File: src/LTO.ts [hunk 3]
```diff
@@ -56,11 +41,27 @@ export class LTO {
...
+    const random = crypto.generateRandomUint32Array(length);
+    const wordCount = dictionary.length;
+    const phrase = [];
+  
+    for (let i = 0; i < length; i++) {
+      const wordIndex = random[i] % wordCount;
+      phrase.push(dictionary[wordIndex]);
+    }
+  
+    random.set(new Uint8Array(random.length));
```
Comment (jasny) [inline src/LTO.ts:55]: What does this line do?

File: package.json [hunk 4]
```diff
@@ -16,19 +16,22 @@
...
   "dependencies": {
     "@lto-network/lto-transactions": "^1.2.7",
+    "@types/secp256k1": "^4.0.3",
     "bignumber.js": "^4.1.0",
     "crypto-js": "^3.1.9-1",
     "js-sha256": "^0.9.0",
     "node-fetch": "^2.2.0",
     "parse-json-bignumber": "0.0.2",
+    "secp256k1": "^4.0.2",
+    "secp256r1": "0.0.3",
```
Comment (jasny) [inline package.json:26]: Not related to this PR.

File: src/classes/transactions/CancelSponsorship.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                "timestamp": this.timestamp,
+                "fee": this.txFee,
+                "proofs": this.proofs
+            }, this.sponsorJson()));
+    }
+
+    fromData(data){
+        var tx = new CancelSponsorship(data['recipient']);
+        tx.type = data.type;
+        'id' in data ? (tx.id = data['id']): (tx.id = "");
```
Comment (jasny) [inline src/classes/transactions/CancelSponsorship.ts:82]:
```suggestion
        tx.id = data.id ?? "";
```

File: src/classes/publicNode.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+              case 9:
+                  return new CancelLease('').fromData(data);
+              case 18:
+                  return new Sponsorship(data['recipient']).fromData(data);
+              case 19:
+                  return new CancelSponsorship(data['recipient']).fromData(data);
+              case 11:
+                  return new MassTransfer('').fromData(data)
+              default:
+                console.error("Transaction type not recognized")
```
Comment (jasny) [inline src/classes/publicNode.ts:65]: Put this in a separate method.

**Review by jasny — CHANGES_REQUESTED** [review]
Also mind the code style.

* Spacing before and after `{` and `}`
* Indenting
* New line after function / code block.
* `} else {` on a single line
* etc

File: src/LTO.ts (hunk 3 above)
Comment (AndreaScorza) [reply src/LTO.ts:55]: I don't see the point of it, it is not a function i coded, it was already there, i will remove it

**Review by jasny — APPROVED** [review]

**Pull request merged**